codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordDecoder.java

```diff
@@ -184,4 +195,17 @@ public static String decodeName(ByteBuf in) {
 
         return name.toString();
     }
+
+    public static String decodeCharString(ByteBuf in) {
+        if (in.readableBytes() == 0) {
+            return null;
+        }
+        int length = (int) in.readByte();
+        if (in.readableBytes() < length) {
```

Comment:
Don't think so.  DefaultDnsRecordDecoder#decodeRecord(ByteBuf) is already doing the check that we have enough data to decode the entire record. Maybe an exception instead?